swupd/helpers_test.go

```diff
@@ -218,3 +200,37 @@ func mustCreateManifests(t *testing.T, ver uint32, minVer bool, format uint, tes
 		t.Fatal(err)
 	}
 }
+
+func checkManifestContains(t *testing.T, testDir, ver, name string, subs []string, expRes bool) {
+	manFpath := filepath.Join(testDir, "www", ver, "Manifest."+name)
+	b, err := ioutil.ReadFile(manFpath)
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	for _, sub := range subs {
+		if bytes.Contains(b, []byte(sub)) != expRes {
+			if expRes {
+				t.Errorf("%s/Manifest.%s did not contain expected '%s'", ver, name, sub)
+			} else {
+				t.Errorf("%s/Manifest.%s contained unexpected '%s'", ver, name, sub)
+			}
+		}
+	}
+}
+
+func checkManifestContainsRes(t *testing.T, testDir, ver, name string, res []*regexp.Regexp) {
+	manFpath := filepath.Join(testDir, "www", ver, "Manifest."+name)
+	b, err := ioutil.ReadFile(manFpath)
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	for _, re := range res {
+		if string(re.Find(b)) == "" {
```

Comment:
`Find` is a bit heavyweight, and then having to convert the array of bytes to a string to compare to the empty string seems to be adding insult to injury!

> if re.Match(b) ==false {